arches/app/datatypes/datatypes.py

```diff
@@ -724,18 +724,19 @@ def set_timezone(self, value):
     def transform_value_for_tile(self, value, **kwargs):
         value = None if value == "" else value
         if value is not None:
-            if type(value) == list:
+            if isinstance(value, list):
                 value = value[0]
-            elif (
-                type(value) == str and len(value) < 4 and value.startswith("-") is False
-            ):  # a year before 1000 but not BCE
-                value = value.zfill(4)
-            if type(value) == str:
+            if isinstance(value, str):
```

Comment:
I know the flax & teal arches orm subclasses string to add localization, so that's a good real-world case for avoiding these exact type checks.